sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -148,6 +149,8 @@ private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
 
 object SharedState {
 
+  URL.setURLStreamHandlerFactory(new SparkUrlStreamHandlerFactory())
```

Comment:
What @jerryshao said.

But I also don't see the need to create any abstraction until it's necessary. So really there's no point in implementing it at this point. If you want to use the hypothetical argument of supporting a new FS, I'll give you the argument that such FS would be implemented as a `FileSystem` and it would automatically hook up to `FsUrlStreamHandlerFactory`, so no need to modify Spark.